tests/components/geo_location/test_demo.py

```diff
@@ -29,59 +30,32 @@ def tearDown(self):
             self.assertTrue(setup_component(self.hass, geo_location.DOMAIN,
                                             CONFIG))
-
-    def test_setup_platform(self):
-        """Test setup of demo platform."""
-        devices = []
-
-        @callback
-        def add_devices_callback(events):
-            """Add recorded devices."""
-            devices.extend(events)
-
-        self.assertTrue(setup_platform(self.hass, None, add_devices_callback,
-                                       None))
-        assert len(devices) == NUMBER_OF_DEMO_DEVICES
-
-    def setup_manager(self):
-        """Setup demo manager."""
-        devices = []
-
-        @callback
-        def add_devices_callback(events):
-            """Add recorded devices."""
-            devices.extend(events)
-
-        return DemoManager(self.hass, add_devices_callback)
-
-    def test_demo_manager(self):
-        """Test demo manager setup."""
-        manager = self.setup_manager()
-        devices = manager._managed_devices
-        assert len(devices) == NUMBER_OF_DEMO_DEVICES
+        entity_ids = self.hass.states.entity_ids(geo_location.DOMAIN).copy()
+        assert len(entity_ids) == NUMBER_OF_DEMO_DEVICES
+        state_first_entry = self.hass.states.get(entity_ids[0])
+        state_last_entry = self.hass.states.get(entity_ids[-1])
         # Check a single device's attributes.
-        device = devices[0]
-        self.assertAlmostEqual(device.distance, device.state, places=0)
-        self.assertAlmostEqual(device.latitude, self.hass.config.latitude,
-                               delta=1.0)
-        self.assertAlmostEqual(device.longitude, self.hass.config.longitude,
-                               delta=1.0)
-        assert device.icon is None
-        assert device.device_state_attributes == {
-            ATTR_LATITUDE: device.latitude, ATTR_LONGITUDE: device.longitude}
-
-    def test_demo_manager_update(self):
-        """Test demo manager setup with update."""
-        manager = self.setup_manager()
-        devices1 = manager._managed_devices.copy()
-        self.assertEqual(NUMBER_OF_DEMO_DEVICES, len(devices1))
+        self.assertAlmostEqual(state_first_entry.attributes['distance'],
+                               float(state_first_entry.state), places=0)
+        self.assertAlmostEqual(state_first_entry.attributes['latitude'],
+                               self.hass.config.latitude, delta=1.0)
+        self.assertAlmostEqual(state_first_entry.attributes['longitude'],
+                               self.hass.config.longitude, delta=1.0)
+        assert state_first_entry.attributes['unit_of_measurement'] == \
+            DEFAULT_UNIT_OF_MEASUREMENT
         # Update (replaces 1 device).
-        manager._update()
-        devices2 = manager._managed_devices.copy()
-        self.assertEqual(NUMBER_OF_DEMO_DEVICES, len(devices2))
-        self.assertNotEqual(devices1, devices2)
+        self.hass.bus.fire(EVENT_TIME_CHANGED,
```

Comment:
You can import the `fire_time_changed` test helper from tests/common.py and use that here.

Reply:
OK, I wasn't aware this exists.